### PR TITLE
Added VSCode Extension Publishing

### PR DESCRIPTION
**Pull request by kuzminrobin**

### CODE REVIEW

File: vscode/.vscodeignore
Comment (billti) [inline vscode/.vscodeignore:5]: Why is this here? You're one directory down (in /vscode), and you're telling it to go up a level and ignore all subdirectories. So.... ignore ALL files in the repo then?

File: vscode/package.json
Comment (billti) [inline vscode/package.json:4]: The 3 lines above this don't see like names or descriptions we'd want to use.

File: .ado/publish.yml
Comment (billti) [inline .ado/publish.yml:1]: Is the name 'publish.yml' required? We're probably going to need publish ADO pipelines for other artifacts too (e.g. pip and npm packages), so the name might want to be more specific if this will be just for the VS Code extension.

File: .ado/publish.yml
Comment (billti) [inline .ado/publish.yml:8]: I assume on each upload the version will need to increment. How does this happen? What is the version numbering scheme?

File: .ado/publish.yml
Comment (billti) [inline .ado/publish.yml:17]: It'll probably build quicker on Linux, unless there's a reason to use the Windows pool.

File: vscode/package.json
Comment (billti) [inline vscode/package.json:51]: Is this a common pattern? I've not seen this before. Being that the license is just standard MIT, I believe this should just be 'MIT'.

**Review by billti — CHANGES_REQUESTED** [review]
See comments.

File: vscode/.vscodeignore
Comment (kuzminrobin) [reply vscode/.vscodeignore:5]: Without this change the `vsce package` was telling that thousands of files will be packed. The `vsce ls` was showing that in addition to the contents of the current dir the whole contents of the upper dir (`../`) will be packed. 

After this change only the files in the current dir were shown as the ones to be backed and `vsce package` did not warn about thousands of files.

File: vscode/package.json
Comment (kuzminrobin) [reply vscode/package.json:4]: These were drafts. I will recover the original lines. 
Feel free to propose new values.

File: .ado/publish.yml
Comment (kuzminrobin) [reply .ado/publish.yml:1]: This .yml is planned for publishing everything (under the same version). And actually, the original sketch contains building the Py packages too, but Py publishing was waiting for the ESRP team approval/reply, so for now I decided to merge the VSCE part only.

After I pass through the whole procedure with ESRP, I'll add Py packages to this file.

File: .ado/publish.yml
Comment (kuzminrobin) [reply .ado/publish.yml:8]: For now the manual update of these lines is assumed (upon release/publish), and manual run of the publishing pipeline. 
After an offline discussion I assume you will give an update.

File: .ado/publish.yml
Comment (kuzminrobin) [reply .ado/publish.yml:17]: I'll try.

File: vscode/package.json
Comment (kuzminrobin) [reply vscode/package.json:51]: Yes, this is a pattern specified [here](https://code.visualstudio.com/api/references/extension-manifest), search for "SEE LICENSE".

File: .ado/publish.yml
Comment (kuzminrobin) [reply .ado/publish.yml]: Wow! In ubuntu it is about 2 times faster. Thanks! :-D

**Comment by kuzminrobin** [comment]
Please review.

File: vscode/package.json
Comment (billti) [inline vscode/package.json:3]: Can we change this to `(preview)` rather than `(new)` please.

**Review by billti — APPROVED** [review]

**Pull request merged**
